src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -209,6 +209,16 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
 
     const [isBackendKilled, setIsBackendKilled] = useState(false);
 
+    const [percentComplete, setPercentComplete] = useState<number | undefined>(undefined);
```

Comment:
I don't understand why this useState and useEffect exists. The state is never read outside the effect, so why not just call `ipcRenderer.send('set-progress-bar', value)` directly in `updateNodeFinish`'s callback?

Reply:
I plan on making a future PR that shows this percentage on the frontend as well. This state will be exported out of the context.